src/compiler/emitter.ts

```diff
@@ -2880,7 +2881,7 @@ namespace ts {
             emitExpression(node.expression, parenthesizer.parenthesizeExpressionOfExpressionStatement);
             // Emit semicolon in non json files
             // or if json file that created synthesized expression(eg.define expression statement when --out and amd code generation)
-            if (!isJsonSourceFile(currentSourceFile!) || nodeIsSynthesized(node.expression)) {
+            if (currentSourceFile && !isJsonSourceFile(currentSourceFile) || nodeIsSynthesized(node.expression)) {

```

Comment:
If you're emitting a purely synthetic expression statement, we should probably emit the `;` by default.

```suggestion
            if (!currentSourceFile || !isJsonSourceFile(currentSourceFile) || nodeIsSynthesized(node.expression)) {
```

Reply:
Ah, yes, I flipped this condition elsewhere and missed this one.